### ISSUE TITLE
Question on MonarchAttention fine-tuning for Bart

### ISSUE THREAD

**Issue by EricLina** [cjyaras/monarch-attention#4]
Hi,

Thank you for your excellent work on monarch-attention!

I noticed that in [ma/ma_triton.py](https://github.com/cjyaras/monarch-attention/blob/main/ma/ma_triton.py), only the forward pass is implemented, and there is currently no backward/autograd support. 

But in your paper, you mention fine-tuning Bart with MonarchAttention. Could you share more details about how you integrated MonarchAttention into Bart for fine-tuning with current triton implmentation? 

Thanks!

**Comment by changwoolee** [comment]
Hi @EricLina 

Thank you for the question! You're right--currently the monarch attention only supports the forward pass. For the Bart experiments, we fine-tuned the Bart model with `softmax attention`. The purpose of fine-tuning was to expose the model to the long sequence data since the original Bart encoder could not process sequences longer than 1024 due to the fixed positional embeddings. After the fine-tuning, we replaced `softmax attention` with `monarch attention`.

Let us know if you have more questions. Thanks!
